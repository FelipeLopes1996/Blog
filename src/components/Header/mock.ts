import { HeaderProps } from '.';

export default {
  blogName: 'Felipe de Lima',
  blogDescription:
    'Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud.',
  logo: '/assets/image/test.jpg',
  showText: true,
} as HeaderProps;
