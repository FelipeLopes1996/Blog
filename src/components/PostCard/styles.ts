import styled, { css } from 'styled-components';

import { Title as HeadingStyle } from '../Heading/styles';

export const Wrapper = styled.div`
  ${({ theme }) => css`
    ${HeadingStyle} {
      margin: 0;
      margin-top: calc(${theme.spacings.small} - 0.6rem);
      margin-bottom: ${theme.spacings.small};
    }
    a {
      text-decoration: none;
      color: inherit;
      transition: opacity 300ms ease-in-out;
    }

    &:hover a {
      color: ${theme.colors.secondary};
    }

    &:hover img {
      opacity: 0.8;
    }
  `}
`;

export const Cover = styled.img`
  max-width: 100%;
  width: 100%;
  transition: opacity 300ms ease-in-out;
`;

export const Excerpt = styled.p``;
