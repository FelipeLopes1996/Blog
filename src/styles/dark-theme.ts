export const darkTheme = {
  title: 'dark',
  colorsDark: {
    primary: '#FFFFFF',
    darkText: '#FFFFFF',
    white: '#000000',
    mediumGray: '#DDDDDD',
    darkerGray: '#AAAAAA',
  },
};
