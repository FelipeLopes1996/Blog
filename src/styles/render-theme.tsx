import { render, RenderResult } from '@testing-library/react';
import React from 'react';
import { BlogThemeProvider } from '../context/changeThema';

export const renderTheme = (children: React.ReactNode): RenderResult => {
  return render(<BlogThemeProvider>{children}</BlogThemeProvider>);
};
