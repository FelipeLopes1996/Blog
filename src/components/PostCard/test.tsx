import { screen } from '@testing-library/react';
import { renderTheme } from '../../styles/render-theme';
import { PostCard, PostCardProps } from '.';

import mock from './mock';

const props: PostCardProps = mock;

describe('<PostCard />', () => {
  it('should render', () => {
    const { container } = renderTheme(<PostCard {...props} />);

    expect(
      screen.getByRole('heading', { name: props.title }),
    ).toBeInTheDocument();
    expect(screen.getByRole('img', { name: mock.title })).toBeInTheDocument();
    expect(screen.getByText(mock.excerpt)).toBeInTheDocument();

    expect(
      screen.getAllByRole('link', { name: mock.title })[0],
    ).toHaveAttribute('href', `/post/${mock.slug}`);

    expect(container).toMatchSnapshot();
  });
});
