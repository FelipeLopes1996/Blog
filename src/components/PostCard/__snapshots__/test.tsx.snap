// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PostCard /> should render 1`] = `
.c3 {
  color: #000000;
  font-size: 2.4rem;
  text-transform: none;
}

.c0 .c2 {
  margin: 0;
  margin-top: calc(1.6rem - 0.6rem);
  margin-bottom: 1.6rem;
}

.c0 a {
  -webkit-text-decoration: none;
  text-decoration: none;
  color: inherit;
  -webkit-transition: opacity 300ms ease-in-out;
  transition: opacity 300ms ease-in-out;
}

.c0:hover a {
  color: #dc143c;
}

.c0:hover img {
  opacity: 0.8;
}

.c1 {
  max-width: 100%;
  width: 100%;
  -webkit-transition: opacity 300ms ease-in-out;
  transition: opacity 300ms ease-in-out;
}

<div>
  <div
    class="c0"
  >
    <a
      href="/post/um-por-do-sol"
    >
      <img
        alt="Um por do sol"
        class="c1"
        src="https://res.cloudinary.com/dgakoaijx/image/upload/v1625254795/9f660f6d0d09d80f6cc039d275567deb_e51239b4d0.png"
      />
    </a>
    <h2
      class="c2 c3"
    >
      <a>
         
        Um por do sol
         
      </a>
    </h2>
    <p
      class=""
    >
      Tudo é bonito quando é de longe fica melhor ainda
    </p>
  </div>
</div>
`;
